src/sentry/integrations/slack/notify_action.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import absolute_import
+
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+from sentry import http
+from sentry.rules.actions.base import EventAction
+from sentry.utils import metrics, json
+from sentry.models import Integration
+
+from .utils import build_attachment
+
+
+class SlackNotifyServiceForm(forms.Form):
+    team = forms.ChoiceField(choices=(), widget=forms.Select(
+        attrs={'style': 'width:150px'},
+    ))
+    channel = forms.CharField(widget=forms.TextInput(
+        attrs={'placeholder': 'i.e #critical-errors'},
+    ))
+
+    def __init__(self, *args, **kwargs):
+        team_list = [(i.id, i.name) for i in kwargs.pop('integrations')]
+        self.channel_transformer = kwargs.pop('channel_transformer')
+
+        super(SlackNotifyServiceForm, self).__init__(*args, **kwargs)
+
+        self.fields['team'].choices = team_list
+        self.fields['team'].widget.choices = self.fields['team'].choices
+
+    def clean_channel(self):
+        channel = self.cleaned_data.get('channel', '').lstrip('#')
+        team = self.cleaned_data.get('team')
+
+        channel_id = self.channel_transformer(team, channel)
+
+        if channel_id is None and team is not None:
+            params = {
+                'channel': channel,
+                'team': dict(self.fields['team'].choices).get(int(team)),
+            }
+
+            raise forms.ValidationError(
+                _('The #%(channel)s channel does not exist in the %(team)s Slack team.'),
+                code='invalid',
+                params=params,
+            )
+
+        return channel
```

Comment:
Originally I had actually been transforming the channel field value here into a slack channel identifier, but it was tricky to transform it back into a channel name for display on the rule (specifically in the form input because of how those values are populated in react).